addons/io_scene_gltf2/blender/exp/gltf2_blender_gather.py

```diff
@@ -32,14 +33,20 @@ def gather_gltf2(export_settings):
     """
     scenes = []
     animations = []  # unfortunately animations in gltf2 are just as 'root' as scenes.
+    materials = []
     active_scene = None
     for blender_scene in bpy.data.scenes:
         scenes.append(__gather_scene(blender_scene, export_settings))
         if export_settings[gltf2_blender_export_keys.ANIMATIONS]:
             animations += __gather_animations(blender_scene, export_settings)
         if bpy.context.scene.name == blender_scene.name:
             active_scene = len(scenes) -1
-    return active_scene, scenes, animations
+
+    for material in bpy.data.materials:
+        double_sided = not material.use_backface_culling
+        materials.append(gltf2_blender_gather_materials.gather_material(material, double_sided, export_settings))
```

Comment:
...to here? The way it works now,`gather_material` (which can be expensive) is called for all materials even if they won't be used.